src/backend/app.js

```diff
@@ -21,6 +23,67 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+// /future-meals: Respond with all meals in the future (relative to the when datetime)
+app.get("/future-meals", async (req, res) => {
+  try {
+    const data = await knex.raw("SELECT `title`, `when` FROM `meal` WHERE `when` > now()");
+    res.status(200).json(data[0]);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+
+// /past-meals:	Respond with all meals in the past (relative to the when datetime)
+app.get("/past-meals", async(req, res) => {
+  try {
+    const data = await knex.raw("SELECT `title`, `when` FROM `meal` WHERE `when` < now()");
+    res.status(200).json(data[0]);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+
+// /all-meals: Respond with all meals sorted by ID
+app.get("/all-meals", async(req, res) => {
+  try {
+    const data = await knex.raw("SELECT `id`, `title` FROM `meal` ORDER BY `id` ASC");
+    res.status(200).json(data[0]);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+
+// /first-meal:	Respond with the first meal (meaning with the minimum id)
+app.get("/first-meal", async(req, res) => {
+  try {
+    const data = await knex.raw("SELECT `id`, `title` FROM `meal` ORDER BY `id` LIMIT 1");
+    
+    data[0].length === 0 
+    ? res.status(404).json("404. The requested URL was not found.")
+    : res.status(200).json(data[0][0]);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+
+// /last-meal:	Respond with the last meal (meaning with the maximum id)
+app.get("/last-meal", async(req, res) => {
+  try {
+    const data = await knex.raw("SELECT `id`, `title` FROM `meal` ORDER BY `id` DESC LIMIT 1");
+    
+    data[0].length === 0
+    ? res.status(404).json("404. The requested URL was not found.")
+    : res.status(200).json(data[0][0]);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("500. Internal Server Error");
+  }
+})
```

Comment:
Nice Job :) 
Your code works fine, but we make the following changes your code will be far easier to read and understand.
- const data --> const meal || const meals
- res.status(404).json("404. The requested URL was not found.")  --> res.status(404).json("404. Meal not found.")
- console.log(error); in every catch block, should be removed.

Reply:
Thank you for the review. I will make the changes you suggested! 🙌